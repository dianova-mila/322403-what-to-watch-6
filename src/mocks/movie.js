export default {
  name: `Macbeth`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  backgroundColor: `#F1E9CE`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  rating: 3.3,
  scoresCount: 48798,
  director: `Justin Kurzel`,
  starring: [
    `Michael Fassbender`,
    `Marion Cotillard`,
    `Jack Madigan`
  ],
  runTime: 113,
  genre: `Drama`,
  released: 2015,
  id: `1`,
  isFavorite: false,
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};
