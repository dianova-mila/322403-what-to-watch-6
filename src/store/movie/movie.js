import {createReducer} from "@reduxjs/toolkit";
import {loadOneMovie, loadPromoMovie, loadComments} from "../actions";

const initialState = {
  movie: {},
  promoMovie: {},
  isPromoMovieLoaded: false,
  comments: [],
  isOneMovieLoaded: false,
};

const movie = createReducer(initialState, (builder) => {
  builder.addCase(loadOneMovie, (state, action) => {
    state.movie = action.payload;
    state.isOneMovieLoaded = true;
  });

  builder.addCase(loadPromoMovie, (state, action) => {
    state.promoMovie = action.payload;
    state.isPromoMovieLoaded = true;
  });

  builder.addCase(loadComments, (state, action) => {
    state.comments = action.payload;
  });
});

export {movie};
