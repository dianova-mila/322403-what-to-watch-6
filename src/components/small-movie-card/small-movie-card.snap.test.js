import React from "react";
import {render} from "@testing-library/react";
import {Router} from "react-router-dom";
import {createMemoryHistory} from "history";
import movie from "../../mocks/movie";
import SmallMovieCard from "./small-movie-card";

it(`SmallMovieCard should render correctly`, () => {
  const history = createMemoryHistory();
  const {container} = render(
      <Router history={history}>
        <SmallMovieCard movie={movie}/>
      </Router>
  );

  expect(container).toMatchSnapshot();
});
