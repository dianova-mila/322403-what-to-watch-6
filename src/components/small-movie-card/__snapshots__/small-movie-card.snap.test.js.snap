// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallMovieCard should render correctly 1`] = `
<div>
  <article
    class="small-movie-card catalog__movies-card"
  >
    <div
      class="small-movie-card__image"
    >
      <img
        alt="Macbeth"
        height="175"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg"
        width="280"
      />
    </div>
    <h3
      class="small-movie-card__title"
    >
      <a
        class="small-movie-card__link"
        href="/films/1/"
      >
        Macbeth
      </a>
    </h3>
  </article>
</div>
`;
